### PR TITLE
refactor: Simplify overall editor layout

### LINKED ISSUE #4257: Text runs over the bottom edge of the page while typing in distraction-free mode.
### Description

When I have Zettlr 3.0.beta.3 in distraction-free mode, the cursor is set below the bottom of the screen and does not move. The text simply runs off the bottom of the page. 

### Reproducing

In the latest beta, enter distraction free mode either through the keyboard shortcut (ctrl+j) or the menu item under "view." 

### Zettlr Version

Beta (if applicable)

### Specify version

3.0.beta.3

### Installation Method

From the Website or GitHub

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Windows 11

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by nathanlesage**
## Description

The current layout relies to a larger-than-healthy amount on JavaScript for calculating sizes, widths, etc. Taking the opportunity opened up by #4257, in this PR I finally perform the overdue housekeeping in simplifying the layout.

## Changes

Everything is a flexbox now.

(Subsequently, there are no more `position: absolute;`s anymore, no weird positioning, etc.)

## Additional information

Still TODO:

- [x] Properly display distraction free mode
- [x] Re-enable "Hide toolbar in distraction free mode"

(Edit: I also allowed myself to fix the `insertImage` on shortcut because inserting an image placeholder when in fact I wanted the DevTools to open drove me nuts)

Fixes #4257

Tested on: macOS Ventura 13.3.1 (design-wise I also tested the `win32` styles)

### CODE REVIEW

File: source/win-main/EditorBranch.vue [hunk 1]
```diff
@@ -198,24 +181,24 @@ body .split-pane-container {
...
-    position: absolute;
     z-index: 1;
     transition: background-color 0.5s ease;
-  }
 
-  .resizer:hover { background-color: rgba(128, 128, 128, .5); }
+    &:hover { background-color: rgba(128, 128, 128, .5); }
+  }
 
   .resizer.horizontal {
```
Comment (kyaso) [inline source/win-main/EditorBranch.vue:192]: We can move this inside the main `.resizer`.

File: source/win-main/EditorBranch.vue [hunk 2]
```diff
@@ -198,24 +181,24 @@ body .split-pane-container {
...
     width: 5px;
+    margin: 0 -2.5px;
     height: 100%;
     cursor: ew-resize;
   }
 
   .resizer.vertical {
```
Comment (kyaso) [inline source/win-main/EditorBranch.vue:199]: (dito)

File: source/win-main/App.vue [hunk 3]
```diff
@@ -41,8 +41,30 @@
...
+              pane that is currently in distraction free mode (if applicable).
+              The other logic below executes in every case, but if a distraction
+              free pane is shown, they only hide. The benefit is that, while the
+              distraction free pane effectively duplicates the pane and thus has
+              a loading time (especially for larger files), this retains
+              whatever state the other panes and branches were in so that the
+              user can - upon leaving distraction free - immediately continue
+              with their work.
+            -->
+            <EditorPane
```
Comment (kyaso) [inline source/win-main/App.vue:57]: In distraction free mode, there seems to be a grey bar appearing on top of the editor:
zettlr-20230422
I did some inspecting, and that bar looks like it's part of this EditorPane, but I couldn't find anything more unfortunately. Also, for me, there is a second vertical scrollbar visible on the right.

File I used: "Pandoc and LaTeX Guide" from the Tutorial.

Does this bar also appear on your machine?

**Review by kyaso — APPROVED** [review]
Looks good to me!

File: source/win-main/EditorBranch.vue (hunk 1 above)
Comment (nathanlesage) [reply source/win-main/EditorBranch.vue:192]: It already is inside the main `.resizer`, I believe …?

File: source/win-main/EditorBranch.vue (hunk 2 above)
Comment (nathanlesage) [reply source/win-main/EditorBranch.vue:199]: Same, or …?

File: source/win-main/App.vue (hunk 3 above)
Comment (nathanlesage) [reply source/win-main/App.vue:57]: Found the issue, there was still a `top: 30px` in the Linux styles.

**Comment by nathanlesage** [comment]
Thanks for finding that one Linux bug! :) Merging …

File: source/win-main/EditorBranch.vue (hunk 1 above)
Comment (kyaso) [reply source/win-main/EditorBranch.vue:189]: If I'm not wrong, it's outside of the main `.resizer`. I saw your change with the `hover` a few lines above, that's how the `.resizer.horizontal` (and `.resizer.vertial`) caught my attention.

Comment (nathanlesage) [reply source/win-main/EditorBranch.vue:189]: Ahhhhh, now I see what you meant: I looked at the changed line, not the one you actually commented on 😂🙈 Yeah, I see your point but I think it's fine for now ... feel free to change that if you think it's nicer that way!

**Pull request merged**